################################################################################
SHORT DESCRIPTION:
################################################################################
Allow non-qdisk CDROM backends in libxl.

################################################################################
LONG DESCRIPTION:
################################################################################
Upstream libxl forces Qdisk as the backend for CD-Rom devices when using
a device model; this is incorrect for OpenXT.

This patch is needed for ATAPI passthrough and to unblock start of guest VMs.

################################################################################
CHANGELOG
################################################################################

################################################################################
REMOVAL
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################

################################################################################
INTERNAL DEPENDENCIES
################################################################################

################################################################################
PATCHES
################################################################################
--- a/tools/libxl/libxl_disk.c
+++ b/tools/libxl/libxl_disk.c
@@ -164,19 +164,6 @@ int libxl__device_disk_setdefault(libxl_
     rc = libxl__resolve_domid(gc, disk->backend_domname, &disk->backend_domid);
     if (rc < 0) return rc;
 
-    /* Force Qdisk backend for CDROM devices of guests with a device model. */
-    if (disk->is_cdrom != 0 &&
-        libxl__domain_type(gc, domid) == LIBXL_DOMAIN_TYPE_HVM &&
-        libxl__device_model_version_running(gc, domid) !=
-        LIBXL_DEVICE_MODEL_VERSION_NONE) {
-        if (!(disk->backend == LIBXL_DISK_BACKEND_QDISK ||
-              disk->backend == LIBXL_DISK_BACKEND_UNKNOWN)) {
-            LOGD(ERROR, domid, "Backend for CD devices on HVM guests must be Qdisk");
-            return ERROR_FAIL;
-        }
-        disk->backend = LIBXL_DISK_BACKEND_QDISK;
-    }
-
     rc = libxl__device_disk_set_backend(gc, disk);
     return rc;
 }
